Index: src/qli/lex.cpp
===================================================================
--- src/qli/lex.cpp	(révision 58885)
+++ src/qli/lex.cpp	(révision 58886)
@@ -438,7 +438,7 @@
 	// UNIX flavor
 
 	if (prompt)
-		printf(prompt);
+		printf("%s", prompt);
 
 	errno = 0;
 	TEXT* p = buffer;
Index: src/dudley/ddl.cpp
===================================================================
--- src/dudley/ddl.cpp	(révision 58885)
+++ src/dudley/ddl.cpp	(révision 58886)
@@ -718,7 +718,7 @@
 	}
 
 	for (;;) {
-		printf(prompt);
+		printf("%s", prompt);
 		if (dudleyGlob.DDL_service)
 			putc('\001', stdout);
 		fflush(stdout);
Index: src/dudley/lex.cpp
===================================================================
--- src/dudley/lex.cpp	(révision 58885)
+++ src/dudley/lex.cpp	(révision 58886)
@@ -429,7 +429,7 @@
 	while (!(c = *DDL_char++)) {
 		DDL_char = DDL_buffer;
 		if (dudleyGlob.DDL_interactive) {
-			printf(dudleyGlob.DDL_prompt);
+			printf("%s", dudleyGlob.DDL_prompt);
 			if (dudleyGlob.DDL_service)
 				putc('\001', stdout);
 			fflush(stdout);
Index: src/lock/print.cpp
===================================================================
--- src/lock/print.cpp	(révision 58885)
+++ src/lock/print.cpp	(révision 58886)
@@ -294,7 +294,7 @@
 			switch (c)
 			{
 			case '?':
-				FPRINTF(outfile, usage);
+				FPRINTF(outfile, "%s", usage);
 				exit(FINI_OK);
 				break;
 
Index: src/gpre/int_cxx.cpp
===================================================================
--- src/gpre/int_cxx.cpp	(révision 58885)
+++ src/gpre/int_cxx.cpp	(révision 58886)
@@ -631,7 +631,7 @@
 	char s[MAX_REF_SIZE];
 
 	align(column);
-	fprintf(gpreGlob.out_file, gen_name(s, action->act_object));
+	fprintf(gpreGlob.out_file, "%s", gen_name(s, action->act_object));
 
 }
 
Index: src/gpre/pat.cpp
===================================================================
--- src/gpre/pat.cpp	(révision 58885)
+++ src/gpre/pat.cpp	(révision 58886)
@@ -458,11 +458,11 @@
 #endif
 
 	default:
-		fprintf(gpreGlob.out_file, buffer);
+		fputs(buffer, gpreGlob.out_file);
 		break;
 	}
 #else
-	fprintf(gpreGlob.out_file, buffer);
+	fputs(buffer, gpreGlob.out_file);
 #endif
 
 }
Index: src/gpre/c_cxx.cpp
===================================================================
--- src/gpre/c_cxx.cpp	(révision 58885)
+++ src/gpre/c_cxx.cpp	(révision 58886)
@@ -1145,7 +1145,7 @@
 			}
 		}
 		*q1 = 0;
-		printa(indent, line);
+		printa(indent, "%s", line);
 		length = length - (q - p);
 		p = q;
 		if (first_line)
@@ -2726,12 +2726,12 @@
 		if (p < limit)
 			continue;
 		*p = 0;
-		printa(INDENT, buffer);
+		printa(INDENT, "%s", buffer);
 		p = buffer;
 	}
 
 	*p = 0;
-	printa(INDENT, buffer);
+	printa(INDENT, "%s", buffer);
 }
 
 
@@ -3499,7 +3499,7 @@
 {
 	TEXT s[MAX_REF_SIZE];
 
-	printa(column, gen_name(s, action->act_object, false));
+	printa(column, "%s", gen_name(s, action->act_object, false));
 }
 
 
Index: src/gpre/pretty.cpp
===================================================================
--- src/gpre/pretty.cpp	(révision 58885)
+++ src/gpre/pretty.cpp	(révision 58886)
@@ -294,7 +294,7 @@
 
 	print_line(control, offset);
 	sprintf(control->ctl_ptr, string, arg);
-	fprintf(stderr, control->ctl_ptr);
+	fputs(control->ctl_ptr, stderr);
 	ADVANCE_PTR(control->ctl_ptr);
 	print_line(control, offset);
 
Index: src/config/Args.cpp
===================================================================
--- src/config/Args.cpp	(révision 58885)
+++ src/config/Args.cpp	(révision 58886)
@@ -177,7 +177,7 @@
 	}
 
 	if (helpText)
-		printf (helpText);
+		printf ("%s", helpText);
 
 	printf ("Options are:\n");
 
@@ -201,7 +201,7 @@
 	for (;;)
 	{
 		if (msg)
-			printf (msg);
+			printf ("%s", msg);
 		printf ("New password: ");
 		if (!fgets (pw1, length, stdin))
 			break;
@@ -241,7 +241,7 @@
 	for (;;)
 	{
 		if (msg)
-			printf (msg);
+			printf ("%s", msg);
 		if (!fgets (pw1, length, stdin))
 			break;
 		char *p = strchr (pw1, '\n');
