--- builds/install/arch-specific/linux/misc/firebird.init.d.mandrake.in	2012/12/14 10:01:27	57484
+++ builds/install/arch-specific/linux/misc/firebird.init.d.mandrake.in	2012/12/14 10:08:41	57485
@@ -99,6 +99,7 @@
 		[ $RETVAL -eq 0 ] && echo "$FULLNAME is running (pid $pid)" || echo "$FULLNAME is dead but pid file exists"
 	else
 		echo "$FULLNAME is stopped"
+        RETVAL=3
 	fi
 	;;
   restart|reload)
