diff -up builds/install/arch-specific/linux/classic/fbembed.pc.in.orig-pc builds/install/arch-specific/linux/classic/fbembed.pc.in
--- builds/install/arch-specific/linux/classic/fbembed.pc.in.orig-pc	2011-11-29 09:05:40.657958483 +0100
+++ builds/install/arch-specific/linux/classic/fbembed.pc.in	2011-11-29 09:11:59.873852393 +0100
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@FB_LIBDIR@
+includedir=@FB_INCDIR@
+
+Name: fbembed
+Description: Firebird embedded client library
+Version: @FIREBIRD_VERSION@
+Cflags: -I${includedir}
+Libs: -L${libdir} -lfbclient
diff -up builds/install/arch-specific/linux/super/fbclient.pc.in.orig-pc builds/install/arch-specific/linux/super/fbclient.pc.in
--- builds/install/arch-specific/linux/super/fbclient.pc.in.orig-pc	2011-11-29 09:05:40.657958483 +0100
+++ builds/install/arch-specific/linux/super/fbclient.pc.in	2011-11-29 09:12:04.660753104 +0100
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@FB_LIBDIR@
+includedir=@FB_INCDIR@
+
+Name: fbclient
+Description: Firebird client library
+Version: @FIREBIRD_VERSION@
+Cflags: -I${includedir}
+Libs: -L${libdir} -lfbclient
diff -up configure.in.orig-pc configure.in
--- configure.in.orig-pc	2011-08-02 17:50:56.000000000 +0200
+++ configure.in	2011-11-29 09:05:40.657958483 +0100
@@ -1290,6 +1290,12 @@ case "$PLATFORM" in
                      gen/install/misc/rc.config.firebird:builds/install/arch-specific/linux/misc/rc.config.firebird.in
                      gen/firebird/bin/linuxLibrary.sh:builds/install/arch-specific/linux/misc/linuxLibrary.sh.in]
       ,[chmod a+x gen/install/scripts/*.sh gen/install/*sh 2>/dev/null])
+    if test $FIREBIRD_ARCH_TYPE = super; then
+      AC_CONFIG_FILES([gen/install/misc/fbclient.pc:${INSTALL_SRC_DIR}/fbclient.pc.in])
+    fi
+    if test $FIREBIRD_ARCH_TYPE = classic; then
+      AC_CONFIG_FILES([gen/install/misc/fbembed.pc:${INSTALL_SRC_DIR}/fbembed.pc.in])
+    fi
     ;;
 
   HPUX)
